src/lib/pubkey/pbes2/pbes2.cpp

```diff
@@ -199,24 +195,21 @@ secure_vector<uint8_t> derive_key(const std::string& passphrase,
 */
 std::pair<AlgorithmIdentifier, std::vector<uint8_t>>
 pbes2_encrypt_shared(std::span<const uint8_t> key_bits,
-                     const std::string& passphrase,
+                     std::string_view passphrase,
                      size_t* msec_in_iterations_out,
                      size_t iterations_if_msec_null,
-                     const std::string& cipher,
-                     const std::string& prf,
+                     std::string_view cipher,
+                     std::string_view prf,
                      RandomNumberGenerator& rng)
    {
-   const std::vector<std::string> cipher_spec = split_on(cipher, '/');
-   if(cipher_spec.size() != 2)
-      throw Encoding_Error("PBE-PKCS5 v2.0: Invalid cipher spec " + cipher);
-
-   if(!known_pbes_cipher_mode(cipher_spec[1]))
-      throw Encoding_Error("PBE-PKCS5 v2.0: Don't know param format for " + cipher);
-
    auto enc = Cipher_Mode::create(cipher, Cipher_Dir::Encryption);
 
-   if(!enc)
-      throw Decoding_Error("PBE-PKCS5 cannot encrypt no cipher " + cipher);
+   const std::vector<std::string> cipher_spec = split_on(cipher, '/');
```

Comment:
Technically, `split_on()` is the perfect candidate to return a vector of `std::string_view`. Maybe, at least use `auto` here, so that line will be agnostic to it.

```suggestion
   const auto cipher_spec = split_on(cipher, '/');
```

Reply:
I'm a bit nervous about returning string_view from anywhere given C++ doesn't really offer the lifetime tracking that Rust does, so you can end up with something like

```
auto uhoh = split_on(some_temporary(), '/')
```

The `auto` is fine though